src/redux/actions/authentication.js

```diff
@@ -1,17 +1,63 @@
-import jwt_decode from "jwt-decode";
-
 import instance from "./instance";
 
-import {} from "./actionTypes";
+import { SET_CURRENT_USER } from "./actionTypes";
 
 import { setErrors } from "./errors";
-
-export const checkForExpiredToken = () => {};
-
-export const login = userData => {};
-
-export const signup = userData => {};
-
-export const logout = () => {};
-
-const setCurrentUser = token => {};
+import decode from "jwt-decode";
+
+export const checkForExpiredToken = () => {
+  const token = localStorage.getItem("token");
+
+  if (token) {
+    const currentTimeInSeconds = Date.now() / 1000;
+
+    const user = decode(token);
+
+    if (user.exp >= currentTimeInSeconds) {
+      return setCurrentUser(token);
+    }
+  }
+
+  return setCurrentUser();
+};
+
+export const login = userData => async dispatch => {
```

Comment:
You could combine these functions

```suggestion
export const authenticate = (userData, type) => async dispatch => {
```